### DISCUSSION TITLE
how to integrate payment gateway midtrans snap pop up

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#650] [Q&A]
at the midtrans payment gateway. i want to use snap pop up feature..so its like html template. which must be run in the browser. how to render html files . so when i click The pay button then moves the page and opens the html file to pay. Any way to render html file in nice gui

Docs here https://docs.midtrans.com/docs/snap

**Comment by falkoschindler** [comment]
Sorry, I don't really understand what you're trying to do. To render an HTML file in NiceGUI, please refer to the documentation about [`ui.html`](https://nicegui.io/documentation/html) and replace the string with arbitrary HTML code loaded from a file.
